From 8fb8dbd5b954cdf5893bc52fe7a2907ee6f8671b Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Thu, 22 Sep 2022 16:08:09 +0800
Subject: [PATCH 021/236] mtk: hostapd: Do not include HE capab IE if
 associated sta's HE capab IE is invalid

The parameter 'sta' passed to send_assoc_resp() might be NULL, so an
NULL check is necessary before access the 'sta'.
Only one such check was missed in this function, and this patch fixs it.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/ieee802_11.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 58ff45918..408c908e1 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -5349,7 +5349,8 @@ static u16 send_assoc_resp(struct hostapd_data *hapd, struct sta_info *sta,
 #endif /* CONFIG_IEEE80211AC */
 
 #ifdef CONFIG_IEEE80211AX
-	if (hapd->iconf->ieee80211ax && !hapd->conf->disable_11ax) {
+	if (hapd->iconf->ieee80211ax && !hapd->conf->disable_11ax && sta &&
+			sta->flags & WLAN_STA_HE) {
 		p = hostapd_eid_he_capab(hapd, p, IEEE80211_MODE_AP);
 		p = hostapd_eid_he_operation(hapd, p);
 		p = hostapd_eid_cca(hapd, p);
-- 
2.45.2

